libsolidity/interface/CompilerStack.cpp

```diff
@@ -533,10 +541,15 @@ bool CompilerStack::compile(State _stopAfter)
 				{
 					try
 					{
-						if (m_generateEvmBytecode)
-							compileContract(*contract, otherCompilers);
-						if (m_generateIR || m_generateEwasm)
+						if (m_viaIR || m_generateIR || m_generateEwasm)
 							generateIR(*contract);
+						if (m_generateEvmBytecode)
+						{
+							if (m_viaIR)
+								generateEVM(*contract);
```

Comment:
```suggestion
								generateEVMFromIR(*contract);
```

Reply:
I kept it parallel to `generateEwasm`. Should we also rename that?